src/statick_tooling/plugins/tool/hadolint_tool_plugin.py

```diff
@@ -0,0 +1,187 @@
+"""Apply hadolint tool and gather results."""
+
+import argparse
+import json
+import logging
+import subprocess
+from typing import List, Optional
+
+from statick_tool.issue import Issue
+from statick_tool.package import Package
+from statick_tool.tool_plugin import ToolPlugin
+
+
+class HadolintToolPlugin(ToolPlugin):  # type: ignore
+    """Apply hadolint tool and gather results."""
+
+    def get_name(self) -> str:
+        """Get name of tool."""
+        return "hadolint"
+
+    def gather_args(self, args: argparse.Namespace) -> None:
+        """Gather arguments."""
+        args.add_argument(
+            "--hadolint-bin",
+            dest="hadolint_bin",
+            type=str,
+            help="hadolint binary path",
+        )
+        args.add_argument(
+            "--hadolint-docker",
+            dest="hadolint_docker",
+            action="store_true",
+            help="Use hadolint docker image instead of binary",
+        )
+
+    # pylint: disable=too-many-locals
+    def scan(self, package: Package, level: str) -> Optional[List[Issue]]:
+        """Run tool and gather output."""
+        tool_bin = "hadolint"
+
+        # If the user explicitly specifies a binary, let that override the default
+        if self.plugin_context and self.plugin_context.args.hadolint_bin is not None:
+            tool_bin = self.plugin_context.args.hadolint_bin
+
+        tool_config = ".hadolint.yaml"
+        user_config = self.plugin_context.config.get_tool_config(
+            self.get_name(), level, "config"
+        )
+        if user_config is not None:
+            tool_config = user_config
+
+        config_file_path = self.plugin_context.resources.get_file(tool_config)
+        flags: List[str] = []
+        flags += ["-f", "json", "--no-fail"]
```

Comment:
```suggestion
        flags: List[str] = ["-f", "json", "--no-fail"]
```

These lines could be combined.